src/invidious/helpers/handlers.cr

```diff
@@ -134,74 +134,6 @@ class AuthHandler < Kemal::Handler
   end
 end
 
-class APIHandler < Kemal::Handler
-  {% for method in %w(GET POST PUT HEAD DELETE PATCH OPTIONS) %}
-  only ["/api/v1/*"], {{method}}
-  {% end %}
-  exclude ["/api/v1/auth/notifications"], "GET"
-  exclude ["/api/v1/auth/notifications"], "POST"
-
-  def call(env)
-    return call_next env unless only_match? env
-
-    env.response.headers["Access-Control-Allow-Origin"] = "*"
-
-    # Since /api/v1/notifications is an event-stream, we don't want
-    # to wrap the response
-    return call_next env if exclude_match? env
-
-    # Here we swap out the socket IO so we can modify the response as needed
-    output = env.response.output
-    env.response.output = IO::Memory.new
```

Comment:
Do IO operations need to be explicitly closed to free up memory in crystal? If yes then this PR will most likely fix a memory leak. (Unless kemal closes it)
https://crystal-lang.org/api/1.10.1/IO/Memory.html#close%3ANil-instance-method